### PR TITLE
add default_ttl to StorageCollection

### LINKED ISSUE #3: No control over ttl in StorageCollection
Hi! 
First of all thanks for this lib, it's really nice.

I encountered small problem. `StorageCollection::retrieve` calls `store` on every storage where key was not found, but you can't set any ttl for it. For Redis it means to store forever, and can be undesirable.

### PR DESCRIPTION
**Pull request by red-led**
closes #3

### CODE REVIEW

**Comment by red-led** [comment]
There's one drawback in this implementation: you can't set different defaults to storage in one collection

File: lib/StorageCollection.php [hunk 1]
```diff
@@ -18,6 +18,18 @@ class StorageCollection extends CacheCollection implements Storage, \ArrayAccess
 {
 	use Storage\ArrayAccess;
 
+	/**
+	 * @var int|null
+	 */
+	private $default_ttl;
+
+	public function __construct(array $collection, $default_ttl = null)
```
Comment (olvlvl) [inline lib/StorageCollection.php]:
```suggestion
	public function __construct(array $collection, int $default_ttl = null)
```

Comment (red-led) [reply lib/StorageCollection.php]: updated with `?int`

**Comment by red-led** [comment]
> Maybe $default_ttl should be a constructor parameter on each storage adapter.

Good idea. How this should work with TTL passed to `store`? I imagine something like `$ttl ?? $this->store` [here](https://github.com/ICanBoogie/Storage/blob/master/lib/RedisStorage.php#L69-L71).

**Comment by olvlvl** [comment]
@red-led Thanks for your contribution.

**Pull request merged**